### DISCUSSION TITLE
About `force_introspection`

### DISCUSSION THREAD

**Discussion by bvr-yr** [hyprwm/Hyprland#4858] [Q&A]
as per wiki
> Introspection is aimed at reducing GPU usage in certain cases, but might cause graphical glitches on nvidia. 

first question, in which cases does it reduce GPU usage?

second, it is a bit misleading: forcing it helps to reduce? or setting it to `0`

I have to set to `1` on intel iGPU cause i see minor glitches in foot (scroll bar) and while some gtk apps launching

I often rely on long battery-time with laptop, so I try to reduce gpu and cpu usage as much as possible when needed. 

what's an impact of setting it 0/1?

**Comment by vaxerski** [comment]
> what's an impact of setting it 0/1?

with introspection you get one additional drawcall at the end of each render pass per damage region

without you dont

**Reply by bvr-yr** [reply]
yeah i see, it brings down gpu usage from ~10% to 5-6% on mpv video with hw-accel

is this a bug that intel affected?
though i didn't noticed any other glitches rather than foot's scrollbar

https://github.com/hyprwm/Hyprland/assets/130279855/a60d3586-96db-4c64-a36d-179884554aaa

**Reply by vaxerski** [reply]
that's odd and doesn't look like a bug from the lack of introspection, tbh

**Reply by bvr-yr** [reply]
hm dunno then
but setting `force_introspection = 1` fixes

i don't know how to file an issue, kinda niche but i doubt it is foot's fault

**Reply by 1player** [reply]
I'm seeing the same bug on foot, but force_introspection = 1 doesn't fix it. AMD GPU.

```
Hyprland, built from branch  at commit e93fbd7c4f991cb8ef03e433ccc4d43587923e15  (props: bump ver to 0.39.0).
Date: Sun Apr 14 18:48:28 2024
Tag: v0.39.0, commits: 1

flags: (if any)
```

**Reply by bvr-yr** [reply]
yeah those glitches still occur, definitely need an opened issue

**Comment by Vaisakhkm2625** [comment]
setting a indicator string fixes the issue in foot

```ini
[scrollback]
indicator-format=#
```

https://man.archlinux.org/man/foot.ini.5.en#SECTION:_scrollback
